jwt-compact does not build with p265 feature on riscv32imac-esp-espidf

## Bug report

We're trying to use jwt-compact together with an esp32c6 (with std support).
Hs256 works just fine (builds, runs, verifies), but we want to use P265 / ES256 instead. 
However, just adding the "p265" feature makes the compilation fail:

```
   Compiling jwt-compact v0.8.0
error[E0599]: no variant or associated item named `ensure_curve` found for enum `JsonWebKey` in the current scope
   --> /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/alg/p256.rs:115:21
    |
115 |         JsonWebKey::ensure_curve(curve, "P-256")?;
    |                     ^^^^^^^^^^^^ variant or associated item not found in `JsonWebKey<'_>`
    |
   ::: /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/jwk.rs:225:1
    |
225 | pub enum JsonWebKey<'a> {
    | ----------------------- variant or associated item `ensure_curve` not found for this enum

error[E0599]: no variant or associated item named `ensure_len` found for enum `JsonWebKey` in the current scope
   --> /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/alg/p256.rs:116:21
    |
116 |         JsonWebKey::ensure_len("x", x, COORDINATE_SIZE)?;
    |                     ^^^^^^^^^^ variant or associated item not found in `JsonWebKey<'_>`
    |
   ::: /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/jwk.rs:225:1
    |
225 | pub enum JsonWebKey<'a> {
    | ----------------------- variant or associated item `ensure_len` not found for this enum

error[E0599]: no variant or associated item named `ensure_len` found for enum `JsonWebKey` in the current scope
   --> /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/alg/p256.rs:117:21
    |
117 |         JsonWebKey::ensure_len("y", y, COORDINATE_SIZE)?;
    |                     ^^^^^^^^^^ variant or associated item not found in `JsonWebKey<'_>`
    |
   ::: /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/jwk.rs:225:1
    |
225 | pub enum JsonWebKey<'a> {
    | ----------------------- variant or associated item `ensure_len` not found for this enum

error[E0599]: no variant or associated item named `ensure_len` found for enum `JsonWebKey` in the current scope
   --> /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/alg/p256.rs:143:21
    |
143 |         JsonWebKey::ensure_len("d", sk_bytes, 32)?;
    |                     ^^^^^^^^^^ variant or associated item not found in `JsonWebKey<'_>`
    |
   ::: /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/jwk.rs:225:1
    |
225 | pub enum JsonWebKey<'a> {
    | ----------------------- variant or associated item `ensure_len` not found for this enum

error[E0599]: no method named `ensure_key_match` found for reference `&JsonWebKey<'_>` in the current scope
   --> /home/leon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jwt-compact-0.8.0/src/alg/p256.rs:147:13
    |
147 |         jwk.ensure_key_match(sk)
    |             ^^^^^^^^^^^^^^^^ method not found in `&JsonWebKey<'_>`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `jwt-compact` (lib) due to 5 previous errors
warning: build failed, waiting for other jobs to finish...
```


### Steps to reproduce
1. Generate a project with https://github.com/esp-rs/esp-idf-template -> Select esp32c6 as target and enable std support (default).
2. Add jwt-compact as `jwt-compact = {version = "0.8.0" }` -> compiles fine
3. Add p265 feature `jwt-compact = {version = "0.8.0", features = ["p256"] }` -> does not compile

### Expected behavior

No compilation errors.

--- a/.github/workflows/build-reusable.yml
+++ b/.github/workflows/build-reusable.yml
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Install packages
         run: |
@@ -30,13 +30,13 @@ jobs:
           toolchain: ${{ inputs.rust_version }}
           components: rustfmt, clippy
       - name: Install cargo-deny
-        uses: baptiste0928/cargo-install@v2
+        uses: baptiste0928/cargo-install@v3
         with:
           crate: cargo-deny
           version: "^0.14"
 
       - name: Cache cargo build
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: target
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
@@ -80,23 +80,21 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Install Rust
         uses: dtolnay/rust-toolchain@master
         with:
           toolchain: stable
       - name: Install Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
-          node-version: '18'
+          node-version: '20'
       - name: Install wasm-pack
-        uses: jetli/wasm-pack-action@v0.4.0
-        with:
-          version: 'latest'
+        run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
 
       - name: Cache cargo build
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: target
           key: ${{ runner.os }}-wasm-cargo-${{ hashFiles('**/Cargo.lock') }}
@@ -121,7 +119,7 @@ jobs:
         working-directory: e2e-tests/no-std
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Install qemu
         run: |
@@ -135,7 +133,7 @@ jobs:
           targets: thumbv7m-none-eabi
 
       - name: Cache cargo build
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: target
           key: ${{ runner.os }}-nostd-cargo-${{ hashFiles('**/Cargo.lock') }}

--- a/.github/workflows/build-reusable.yml
+++ b/.github/workflows/build-reusable.yml
@@ -56,6 +56,8 @@ jobs:
         run: cargo clippy -p jwt-compact --no-default-features --features std,ed25519-compact --all-targets -- -D warnings
       - name: Clippy k256
         run: cargo clippy -p jwt-compact --no-default-features --features k256 --all-targets -- -D warnings
+      - name: Clippy p256
+        run: cargo clippy -p jwt-compact --no-default-features --features p256 --all-targets -- -D warnings
       - name: Clippy WASM crate
         run: cargo clippy -p jwt-compact-wasm --all-targets -- -D warnings
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,7 +16,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Install packages
         run: |
@@ -28,7 +28,7 @@ jobs:
           toolchain: ${{ env.msrv }}
 
       - name: Cache cargo build
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: target
           key: ${{ runner.os }}-msrv-cargo-${{ hashFiles('**/Cargo.lock') }}
@@ -58,7 +58,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           persist-credentials: false
 
@@ -72,7 +72,7 @@ jobs:
           toolchain: ${{ env.nightly }}
 
       - name: Cache cargo build
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: target
           key: ${{ runner.os }}-document-cargo-${{ hashFiles('**/Cargo.lock') }}

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,9 +5,13 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
 ## [Unreleased]
 
-## Changed
+### Changed
 
 - Bump minimum supported Rust version to 1.70.
+
+### Fixed
+
+- Fix missing JWK helpers when compiling the crate with the `p256` feature.
 
 ## 0.8.0 - 2023-10-12
 
@@ -75,17 +79,13 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
 - Add `UntrustedToken::into_owned()` method to extend the token lifetime to static.
   This is useful if an `UntrustedToken` needs to be stashed / passed across threads.
-
 - Add `UntrustedToken::deserialize_claims_unchecked()` to extract claims from a token
   without verification. As the name length implies, the method should only be used
   in exceptional cases.
-
 - Implement `FromStr` and some standard traits (`Clone`, `Copy`, `PartialEq`) for `Rsa`.
-
 - Introduce `AlgorithmSignature::LENGTH` constant for specifying the expected 
   signature length. The new `ValidationError::InvalidSignatureLen` variant provides
   more specific errors if the signature length is not as specified.
-
 - Allow opting out from CBOR claims encoding by making `serde_cbor` dependency optional
   and using it as a crate feature. The relevant functionality (`AlgorithmExt::compact_token`,
   some error variants, etc.) is now gated behind this feature.
@@ -101,7 +101,6 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 ### Fixed
 
 - Fix datetime overflow when validating the expiration claim.
-
 - Fix `no_std` support for RSA-based JWS algorithms. As a part of the fix,
   to enable RSA, you should now use the `rsa` feature instead of `rsa`.
 
@@ -115,28 +114,20 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
 - Add `new()` constructor for `HS*` keys that accepts any type implementing
   `AsRef<[u8]>`. This simplifies key loading.
-
 - Add `new()` constructor for `UntrustedToken` that accepts any type implementing
   `AsRef<str>`. This simplifies token processing.
-
 - Add basic JSON Web Key (JWK) support. This allows (de)serializing keys from / into
   a uniform format and computing key thumbprints.
-
 - Add ES256K implementation using pure-Rust [`k256`] crate.
 
 ### Changed
 
 - Update dependencies.
-
 - Rename `Header.signature_type` field to `token_type` to be more in line with JWT spec.
-
 - Rename `Claims.expiration_date` to `expiration` to be more precise.
-
 - Encapsulate `hmac` and `sha2` dependencies by introducing signature types 
   for `HS*` algorithms. The `digest` dependency is still public.
-
 - `exonum-crypto` feature is no longer enabled by default.
-
 - Change return type of `SigningKey::as_bytes()` to securely zeroize owned values on drop.
 
 ### Fixed
@@ -155,7 +146,6 @@ No substantial changes compared to the 0.3.0-beta.2 release.
 ### Changed
 
 - Make `CreationError` non-exhaustive.
-
 - Rename `StrongKey::inner()` method to `into_inner`.
 
 ### Fixed
@@ -167,33 +157,23 @@ No substantial changes compared to the 0.3.0-beta.2 release.
 ### Added
 
 - Add signature getters for untrusted and trusted tokens.
-
 - Add the [`ed25519-compact`] backend for Ed25519-based tokens.
-
 - Add `SigningKey` and `VerifyingKey` traits for generic access to cryptographic keys.
-
 - Support RSA algorithms using pure-Rust [`rsa`] crate.
-
 - Add `no_std` mode and check `no_std` / WASM compatibility via dedicated aux crates.
   Introduce two relevant crate features, `clock` and `std`.
-
 - Add wrapper types for strong keys / JWT algorithms.
-
 - Add details to some `ValidationError` variants.
 
 ### Changed
 
 - Update dependencies.
-
 - Update minimum supported Rust version due to dependencies.
-
 - `es256k` feature should now be used for access to libsecp256k1 backend instead of
   `secp256k1`.
-
 - Rework time-related token creation / validation logic. It is now possible to
   use a custom clock, which could be useful for testing or if there is no access
   to the system clock.
-
 - Make `Header`, `Claims`, `TimeOptions`, and error types non-exhaustive.
 
 ## 0.2.0 - 2020-05-11

--- a/src/jwk.rs
+++ b/src/jwk.rs
@@ -419,6 +419,7 @@ pub struct RsaPrimeFactor<'a> {
 #[cfg(any(
     feature = "es256k",
     feature = "k256",
+    feature = "p256",
     feature = "exonum-crypto",
     feature = "ed25519-dalek",
     feature = "ed25519-compact"

